app/jobs/task_job.rb

```diff
@@ -0,0 +1,22 @@
+class TaskJob < ApplicationJob
+  before_perform do |job|
+    task_for_job(job).start
+  end
+
+  around_perform do |job, perform_block|
+    perform_block.call
```

Comment:
You can cut the `before_perform` if you move `task_for_job(job).start` to the `around_perform` hook before the block call.